### PR TITLE
Update Release Notes template

### PR DESCRIPTION
**Pull request by ricardoV94**
Towards #115 

Sister PR in https://github.com/pymc-devs/pymc/pull/6392

### CODE REVIEW

File: .github/release.yml [hunk 1]
```diff
@@ -8,19 +8,12 @@ changelog:
...
-    - title: Docs & Maintenance 🔧
-      labels:
-        - docs
-        - installation
-        - maintenance
-        - pre-commit
-        - tests
-        - "*"
\ No newline at end of file
+        - aesara downstream
```
Comment (michaelosthege) [inline .github/release.yml:19]: IMO we should keep the existing categories and just add a "Changes in Aesara that were incorporated in PyTensor" section.

It's fine if things are listed twice, and I think we should do better than the Aesara developers when it comes to writing informative release notes. Separating changes from maintenance or improvements is an important part of this.

**Review by michaelosthege — CHANGES_REQUESTED** [review]

File: .github/release.yml (hunk 1 above)
Comment (ricardoV94) [reply .github/release.yml:19]: Do the automatic release notes work with duplicated matches?

Comment (michaelosthege) [reply .github/release.yml:19]: https://github.com/pymc-devs/pymc/pull/6375 had "maintenance" and "major" labels and ended up in the "major" section.

So I guess the first category matching a PR is the one where it's going to be listed.

You could place the "Changes in Aesara that were incorporated in PyTensor" first in the template and then we only need to curate to move it down & to copy items to the "Major" section

---------------

Maybe we should look at a completely different approach: How about prefixing the downstreaming PRs with `🔄 From Aesara: ` such that the bullet points in the release notes can still be categorized by function/impact?

Comment (ricardoV94) [reply .github/release.yml:19]: Yeah, maybe that's better

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**